avocado/core/nrunner.py

```diff
@@ -139,6 +139,34 @@ def pick_runner_command(task, runners_registry=None):
     runners_registry[kind] = False
 
 
+def check_tasks_requirements(tasks, runners_registry=None):
+    """
+    Checks if tasks have runner requirements fulfilled
+
+    :param tasks: the tasks whose runner requirements will be checked
+    :type tasks: list of :class:`avocado.core.nrunner.Task`
+    :param runners_registry: a registry with previously found (and not
+                             found) runners keyed by task kind
+    :param runners_registry: dict
+    :return: two list of tasks in a tupple, with the first being the tasks
+             that pass the requirements check and the second the tasks that
+             fail the requirements check
+    :rtype: tupple of (list, list)
+    """
+    if runners_registry is None:
+        runners_registry = KNOWN_RUNNERS
+
+    ok = []
+    missing = []
+    for task in tasks:
+        runner = pick_runner_command(task, runners_registry)
+        if runner:
```

Comment:
Also in this case, we could create a check method in the `Task` class? Because looks like a loop like this would be cleaner:

```python
for task in tasks:
  if task.is_valid(): # or other name
    ok.append(task)
  else:
    missing.append(task)
```

What do you think?

Reply:
I'm going with `Task.are_requirements_available()` because `is_valid()` doesn't tell the whole story. A task may be valid in the sense that it has all the right info, but a system can't run it.